src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,411 @@
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** When `true`, focuses the `textarea` element on page render. */
+  @Prop({ reflect: true }) autofocus = false;
+
+  /** When `true`, disables the component. */
+  @Prop({ reflect: true }) disabled = false;
+
+  /** Specifies the placeholder text for the input. */
+  @Prop() placeholder: string;
+
+  /** Whne `true`, the component's value can be read, but cannot be modified.  */
+  @Prop({ reflect: true }) readonly = false;
+
+  /** Specifies number or rows allowed. */
+  @Prop({ reflect: true }) rows: number;
+
+  /** Specifies number or columns allowed. */
+  @Prop({ reflect: true }) cols: number;
+
+  /** Specifies maximum number of characters allowed. */
+  @Prop() maxlength: number;
+
+  /** Specifies name of the component  */
+  @Prop({ reflect: true }) name: string;
+
+  /** Specifies the size of `textarea` component. */
+  @Prop({ reflect: true }) scale: "l" | "m" | "s" = "m";
+
+  /** Specifies wrapping mechanism for the text.  */
+  @Prop({ reflect: true }) wrap: "soft" | "hard" = "soft";
+
+  /** The component's value. */
+  @Prop({ mutable: true }) value: string;
+
+  /** When `true`, disables the resizing handle. */
+  @Prop({ reflect: true }) resizeDisabled = false;
+
+  /** When `true`, disables resizing textarea horizantally. */
+  @Prop({ reflect: true }) horizantalResizeDisabled = false;
+
+  /** When `true`, disables resizing textarea vertically. */
+  @Prop({ reflect: true }) verticalResizeDisabled = false;
+
+  /** When `true`, marks this component as required in form. */
+  @Prop({ reflect: true }) required = false;
+
+  /** The label of the component */
+  @Prop() label: string;
+
+  /** When true, the `textarea` will be marked as invalid. */
+  @Prop({ reflect: true }) invalid = false;
+
+  /**
+   * Specifies the Unicode numeral system used by the component for localization.
+   */
+  @Prop() numberingSystem: NumberingSystem;
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  @Prop({ mutable: true }) messages: TextareaMessages;
+
+  /**
+   * Use this property to override individual strings used by the component.
+   */
+  @Prop({ mutable: true }) messageOverrides: Partial<TextareaMessages>;
+
+  @Watch("messageOverrides")
+  onMessagesChange(): void {
+    /* wired up by t9n util */
+  }
+
+  @Watch("disabled")
+  disabledHandler(value: boolean): void {
+    this.disablePointerEvents(value, this.leadingSlotElements);
+    this.disablePointerEvents(value, this.trailingSlotElements);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Emitters
+  //
+  //--------------------------------------------------------------------------
+  /**
+   * Fires each time a new `value` is typed.
+   */
+  @Event({ cancelable: true }) calciteTextareaInput: EventEmitter<void>;
```

Comment:
Initially i did considered naming `calcite-text-area` but to keep it in sync with the native element chose `calcite-textarea`. I dont have a strict preference but `calcite-text-area` is a bit odd :thinking:

Reply:
@macandcheese @eriklharper any thoughts here? We have deviated from the native naming pattern of smushing two words together in the past, although it was related to property not element names. Here is an example: https://github.com/Esri/calcite-components/pull/3382#issuecomment-962253064

Native can't hyphenate element names, but as custom elements we can, so in my opinion we might as well.

> [Custom elements] contain a hyphen, used for namespacing and to ensure forward compatibility __(since no elements will be added to HTML, SVG, or MathML with hyphen-containing local names in the future)__.
   -- [spec](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name)

Reply:
Native tag is <textarea>, this is a replication of that element so matching that makes sense.

Reply:
I lean towards `textarea` too for the same reasons Adam mentioned.  I actually also thought as a pattern that we wanted all our "input" components to prefix the word "input" in front of them, but maybe that only applies to the native `<input type="type">` components we're breaking away from `input` and not `textarea`.

Reply:
`calcite-input-textarea` is longer, but does have more semantic meaning.

Reply:
Yeah we shouldn't put `input` in the name since the whole point is breaking it out. It was between `calcite-text-area` and `calcite-textarea`. Sounds like both Erik and Adam vote for `calcite-textarea` so we can stick with that

Reply:
I've had similar conversations with my teams regarding attributes such as `autocomplete` where we decided to keep it as to remain less confusing when external developers attempt to apply the attributes in the way they are used to doing so. In regards to component naming though, I would like to add to the argument for separation due to the fact they are already different via the `calcite-` prefix (i.e. `calcite-textarea` vs `textarea` so taking this one step further with `calcite-text-area` shouldn't really be any more confusing :)).